test/tests_for_utils/job_requirements_resolver_test.py

```diff
@@ -645,3 +646,446 @@ def test_get_configured_client_group_spec_fail():
+
+    assert jrr.resolve_requirements(
+        " module2. some_meth  ", client_group="cg2"
+    ) == JobRequirements(
+        4,
+        2000,
+        100,
+        "cg1",
+    )
+
+    catalog.list_client_group_configs.assert_called_once_with(
+        {"module_name": "module2", "function_name": "some_meth"}
+    )
+
+
+def test_resolve_requirements_from_catalog_full_CSV():
+    catalog = create_autospec(Catalog, spec_set=True, instance=True)
+    catalog.list_client_group_configs.return_value = [
+        {
+            "client_groups": [
+                "cg1",
+                "request_cpus=  78",
+                "   request_memory   = 500MB",
+                "request_disk = 700GB",
+                "client_group_regex = False",
+                "debug_mode = true",
+                "foo=bar",
+                "baz=bat",
+            ]
+        }
+    ]
+
+    spec = _get_simple_deploy_spec_file_obj()
+
+    jrr = JobRequirementsResolver(catalog, spec)
+
+    assert jrr.resolve_requirements(" module2. some_meth  ") == JobRequirements(
+        78,
+        500,
+        700,
+        "cg1",
+        False,
+        None,
+        False,
+        {"foo": "bar", "baz": "bat"},
+        True,
+    )
+
+    catalog.list_client_group_configs.assert_called_once_with(
+        {"module_name": "module2", "function_name": "some_meth"}
+    )
+
+
+def test_resolve_requirements_from_catalog_partial_JSON():
+    catalog = create_autospec(Catalog, spec_set=True, instance=True)
+    catalog.list_client_group_configs.return_value = [
+        {
+            "client_groups": [
+                '{"client_group": "  cg1  "',
+                '"    request_memory    ":  "  300M   "',
+                '"exactlythesameshape": "asathingy"',
+                '"request_disk": 100000}',
+            ]
+        }
+    ]
+
+    spec = _get_simple_deploy_spec_file_obj()
+
+    jrr = JobRequirementsResolver(catalog, spec)
+
+    assert jrr.resolve_requirements(" module2. some_meth  ") == JobRequirements(
+        4,
+        300,
+        100000,
+        "cg1",
+        scheduler_requirements={"exactlythesameshape": "asathingy"},
+    )
+
+    catalog.list_client_group_configs.assert_called_once_with(
+        {"module_name": "module2", "function_name": "some_meth"}
+    )
+
+
+def test_resolve_requirements_from_user_full():
+    _resolve_requirements_from_user_full(True)
+    _resolve_requirements_from_user_full(False)
+
+
+def _resolve_requirements_from_user_full(bool_val):
+    catalog = create_autospec(Catalog, spec_set=True, instance=True)
+    catalog.list_client_group_configs.return_value = [
+        {
+            "client_groups": [
+                "cg2",
+                "request_cpus=  78",
+                "   request_memory   = 500MB",
+                "request_disk = 700GB",
+                "client_group_regex = False",
+                "debug_mode = true",
+                "foo=bar",
+                "baz=bat",
```

Comment:
how about a value with some extra `=` signs, just to test that splitting operation?

`foo = bar=baz=bat=whatever`

Reply:
fixed